springboot/src/main/java/org/mobilitydata/gtfsvalidator/springboot/GtfsValidatorController.java

```diff
@@ -0,0 +1,329 @@
+  private static final String MESSAGE_JSON_KEY = "message";
+  private static final String DEFAULT_COUNTRY_CODE = "ZZ";
+
+  /**
+   * Runs {@code org.mobilitydata.gtfsvalidator.cli.Main.main} with the arguments extracted via the
+   * query parameters. Returns the the {@code ResponseEntity} that contains information about the
+   * response's {@code HttpStatus} and a message (as a {@code String}) that gives more information
+   * about success or failure. Please note that this method requires authentication to be setup
+   * prior to execution i.e. environment variable GOOGLE_APPLICATION_CREDENTIALS AND
+   * VALIDATION_REPORT_BUCKET have to be defined.
+   *
+   * <p>Please check:
+   *
+   * <ul>
+   *   <li><a href=https://cloud.google.com/storage/docs/naming-buckets>this documentation for
+   *       bucket naming guidelines</a>
+   *   <li><a
+   *       href=https://cloud.google.com/docs/authentication/getting-started#setting_the_environment_variable>this
+   *       documentation for authentication procedure</a>
+   * </ul>
+   *
+   * <a href=https://cloud.google.com/storage/docs/json_api/v1/status-codes>Possible HTTP status
+   * codes</a>
+   *
+   * @param output_base Base directory to store the outputs
+   * @param threads Number of threads to use
+   * @param country_code Country code of the feed, e.g., `nl`. It must be a two-letter country code
+   *     (ISO 3166-1 alpha-2)")
+   * @param url The fully qualified URL to download GTFS archive
+   * @param validation_report_name The name of the validation report including .json extension.
+   * @param dataset_id the id of the dataset validated
+   * @param commit_sha the commit SHA
+   * @return the {@code ResponseEntity} that contains information about the response's {@code
+   *     HttpStatus} and a message (as a {@code String}) that gives more information about success
+   *     or failure. <a href=https://cloud.google.com/storage/docs/json_api/v1/status-codes>Possible
+   *     HTTP status codes</a>
+   */
+  @GetMapping(value = "/", produces = "application/json; charset=UTF-8")
+  @ResponseBody
+  public ResponseEntity<String> run(
+      @RequestParam(required = false, defaultValue = DEFAULT_OUTPUT_BASE) String output_base,
+      @RequestParam(required = false, defaultValue = DEFAULT_NUM_THREADS) String threads,
+      @RequestParam(required = false, defaultValue = DEFAULT_COUNTRY_CODE) String country_code,
+      @RequestParam() String url,
+      @RequestParam(required = false, defaultValue = Arguments.VALIDATION_REPORT_NAME_JSON)
+          String validation_report_name,
+      @RequestParam(required = false, defaultValue = Arguments.SYSTEM_ERRORS_REPORT_NAME_JSON)
+          String system_error_report_name,
+      @RequestParam(required = false, defaultValue = "dataset id value") String dataset_id,
+      @RequestParam(required = false, defaultValue = "commit sha value") String commit_sha) {
+
+    Arguments args =
```

Comment:
Do you still need this step to convert to JCommander Arguments? The only potential benefit I could see is that maybe it's doing some kind of validation, but looking at the docs quickly it doesn't look like it does by default:
https://jcommander.org/#_parameter_validation

So I think it can be removed and remove the dependency too from build.gradle.

Reply:
I think that we should keep it since we use `Arguments` and its methods in the code below.

Reply:
Ah, ok - I thought that Arguments was a JCommander library class, but I see it's not now. It's still a little strange IMHO that the web implementation has an indirect dependency on the CLI arguments library (JCommander), but I see why you're doing it (to avoid duplicating code to hold parameter data).